### PR TITLE
for chgres_cube - soil parameter references in program_setup.f90, turned on doxygen documentation for private functions

### LINKED ISSUE #350: Document soil param calculations in chgres_cube.fd/program_setup.f90
Is there a paper or other reference for the soil parameter calculations in chgres_cube.fd/program_setup.f90?

### PR DESCRIPTION
**Pull request by edwardhartnett**
Fixes #350 
Part of #191 

Add some documentation about where soil parameter calculations came from.

I also turned on documentation for private subprograms. We need to document these as well as the public ones.

### CODE REVIEW

File: sorc/chgres_cube.fd/program_setup.f90 [hunk 1]
```diff
@@ -443,6 +445,25 @@ end subroutine get_var_cond
 !> Driver routine to compute soil parameters for each
 !! soil type. Works for Zobler and STATSGO soil categories.
 !!
+!! The calculations are those used in the Noah Land Surface Model. For
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/program_setup.f90:448]: I can build your branch with no warnings.  But the hyperlinks are not displaying right.  Some documentation I found said the link should be enclosed in ( ) and the wording in the square brackets.

Comment (edwardhartnett) [reply sorc/chgres_cube.fd/program_setup.f90:448]: Stand by, I got that backwards. I will fix...

Comment (edwardhartnett) [reply sorc/chgres_cube.fd/program_setup.f90]: OK, the href tag from html seems to work best. I have changed to that.

**Comment by edwardhartnett** [comment]
@GeorgeGayno-NOAA I think this one is good to merge, but we need you to approve the code review.

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA I think this one is good to merge, but we need you to approve the code review.

I will check now.

**Review by GeorgeGayno-NOAA — APPROVED** [review]
The hyperlinks look good.  Ready to merge.

**Pull request merged**